packages/ts/react-grid/test/autoform.spec.tsx

```diff
@@ -0,0 +1,155 @@
+  field.dispatchEvent(new CustomEvent('input'));
+  await nextFrame();
+}
+async function assertFormFieldValue(result: RenderResult, fieldLabel: string, expected: number | string) {
+  const field = await getFormField(result, fieldLabel);
+  expect(field.value).to.equal(expected);
+}
+async function submit(result: RenderResult) {
+  const submitButton = await result.findByText('Submit');
+  submitButton.click();
+}
+
+describe('@hilla/react-grid', () => {
+  describe('Auto form', () => {
+    it('renders fields for the properties in the form', async () => {
+      const person: Person = {
+        id: 1,
+        firstName: 'first',
+        lastName: 'last',
+        email: 'first.last@domain.com',
+        someNumber: 24451,
+        vip: false,
+      };
+      const result = render(<AutoForm service={personService} model={PersonModel} item={person} />);
+      await assertFormFieldValue(result, 'First name', person.firstName);
+      await assertFormFieldValue(result, 'Last name', person.lastName);
+      await assertFormFieldValue(result, 'Email', person.email);
+      await assertFormFieldValue(result, 'Some number', person.someNumber);
+    });
+    it('works without an item', async () => {
+      const result = render(<AutoForm service={personService} model={PersonModel} />);
+      await assertFormFieldValue(result, 'First name', '');
+      await assertFormFieldValue(result, 'Last name', '');
+      await assertFormFieldValue(result, 'Email', '');
+      await assertFormFieldValue(result, 'Some number', 0);
+    });
+    it('uses values from an item', async () => {
+      const person = (await getItem(personService, 2))!;
+
+      const result = render(<AutoForm service={personService} model={PersonModel} item={person} />);
+      await assertFormFieldValue(result, 'First name', person.firstName);
+      await assertFormFieldValue(result, 'Last name', person.lastName);
+      await assertFormFieldValue(result, 'Email', person.email);
+      await assertFormFieldValue(result, 'Some number', person.someNumber);
+    });
+    it('submits a valid form', async () => {
+      const service: CrudService<Person> & HasLastFilter = createService<Person>(personData);
+
+      const person = await getItem(service, 1);
+
+      const result = render(<AutoForm service={service} model={PersonModel} item={person} />);
+      await setFormField(result, 'First name', 'foo');
+      await submit(result);
+      await nextFrame();
+      const newItem = await getItem(service, 1);
+      expect(newItem!.firstName).to.equal('foo');
+    });
+
+    it('calls onSubmit with the new item', async () => {
+      const service: CrudService<Person> & HasLastFilter = createService<Person>(personData);
+
+      const person = await getItem(service, 1);
+      let submittedItem: Person | undefined;
+      const result = render(
+        <AutoForm
+          service={service}
+          model={PersonModel}
+          item={person}
+          onSubmit={({ item }) => {
```

Comment:
Nit: Could use a spy here.

Reply:
Done

Reply:
I think you missed to update it in this specific test.

Reply:
done